corehq/apps/geospatial/reports.py

```diff
@@ -49,52 +50,30 @@ def template_context(self):
 
         return context
 
-    @property
-    def report_context(self):
-        cases = []
-        invalid_geo_cases_count = 0
-        for row in self.es_results['hits'].get('hits', []):
-            es_case = self.get_case(row)
-            display = CaseDisplayES(es_case, self.timezone, self.individual)
-
-            coordinates = _get_geo_location(es_case)
-            if coordinates is None:
-                invalid_geo_cases_count += 1
-                continue
-            # We should consider passing in a "center_coordinates" fields to center the map
-            # to the relavent
-            case = {
-                "case_id": display.case_id,
-                "case_type": display.case_type,
-                "name": display.case_name,
-                "coordinates": coordinates
-            }
-            cases.append(case)
-
-        invalid_cases_link = self._invalid_geo_cases_report_link if invalid_geo_cases_count else ''
-
-        return dict(
-            context={
-                "cases": cases,
-                "invalid_geo_cases_report_link": invalid_cases_link,
-            },
-        )
-
     @property
     def default_report_url(self):
         return reverse('geospatial_default', args=[self.request.project.name])
 
     @property
-    def _invalid_geo_cases_report_link(self):
-        # Copy the set of filters to pre-populate the Case List Explorer page's filters
-        query = self.request.GET.copy()
-        if 'search_query' in query:
-            query.pop('search_query')
-
-        query['search_xpath'] = f"{GEO_POINT_CASE_PROPERTY} = ''"
-        cle = CaseListExplorer(self.request, domain=self.domain)
-
-        return "{resource}?{query_params}".format(
-            resource=cle.get_url(self.domain),
-            query_params=query.urlencode(),
+    def headers(self):
+        from corehq.apps.reports.datatables import DataTablesColumn, DataTablesHeader
+        headers = DataTablesHeader(
+            DataTablesColumn(_("case_id"), prop_name="type.exact"),
+            DataTablesColumn(_("GPS"), prop_name="type.exact"),
+            DataTablesColumn(_("Name"), prop_name="name.exact", css_class="case-name-link"),
         )
+        headers.custom_sort = [[2, 'desc']]
+        return headers
+
+    @property
+    def rows(self):
+        cases = []
+        for row in self.es_results['hits'].get('hits', []):
+            display = CaseDisplayES(self.get_case(row), self.timezone, self.individual)
```

Comment:
Not a blocker on this PR, but I'm just wondering what should happen if the `get_case(row)` method fails. Currently the request would fail (at least the `self.rows` routine); is this the expected behaviour or can/should we load the cases partially and report to the user that some cases wasn't loaded? 

Maybe this was discussed at some point in the past, but if not I'd like to bring that to attention.

Reply:
That's a good point. I believe the exceptions raised by `get_case` are on completely unexpected/inconsistent document state which are good be failing hard on. The CaseListReport also uses `get_case` without catching the exceptions (I think for the same reason)